### PR TITLE
Revert #4581 and fix #4480 by using a temporary file with a redirect to the correct location.

### LINKED ISSUE #4480: `help SECTION` doesn't work on macOS
On git master on macOS, `help variables` (or any section) no longer takes you to the relevant section. It just takes you to index.html without an anchor.

### PR DESCRIPTION
**Pull request by thomcc**
## Description

This implements the fixes described in [my comment here](https://github.com/fish-shell/fish-shell/issues/4480#issuecomment-357353835).

## TODOs:

- [ ] Changes to fish usage are reflected in user documentation/manpages.
- [ ] Tests have been added for regressions fixed
- [ ] User-visible changes noted in CHANGELOG.md

A regression test for this doesn't really seem practical to me, but I've done a bunch of manual testing on this to make sure it works.

### CODE REVIEW

**Comment by faho** [comment]
The code seems alright to me, but I'd like someone else on macOS to sign off on this.

@floam, @ridiculousfish, @zanchey: Does this work?

File: share/functions/help.fish [hunk 1]
```diff
@@ -127,17 +132,24 @@ function help --description 'Show help for the fish shell'
...
+        set need_trampoline
     end
 
-    # OS X /usr/bin/open swallows fragments (anchors), so use osascript
-    # Eval is just a cheesy way of removing the hash escaping
-    if test "$fish_browser" = osascript
-        set -l opencmd 'open location "'(eval echo $page_url)'"'
-        osascript -e 'try' -e $opencmd -e 'on error' -e $opencmd -e 'end try'
-        return
+    if set -q need_trampoline[1]
```
Comment (floam) [inline share/functions/help.fish]: Could we use `psub` to get a temporary file that fish deletes for us for free, simplifying the code?

File: share/functions/help.fish [hunk 2]
```diff
@@ -127,17 +132,24 @@ function help --description 'Show help for the fish shell'
...
+        if set -l clean_url (string replace '\\#' '#' $page_url)
+            # Write a temporary file that will redirect where we want.
+            set -q TMPDIR
+            or set -l TMPDIR /tmp
+            set -l tmpdir (mktemp -d $TMPDIR/help.XXXXXX)
+            set -l tmpname $tmpdir/help.html
+            echo '<meta http-equiv="refresh" content="0;URL=\''$clean_url'\'" />' > $tmpname
+            set page_url file://$tmpname
+        end
     end
```
Comment (floam) [inline share/functions/help.fish]: v

**Comment by floam** [comment]
Works fine on MacOS.

File: share/functions/help.fish (hunk 2 above)
Comment (thomcc) [reply share/functions/help.fish]: That doesn't seem to work, since `open` returns before launching the browser.

**Comment by faho** [comment]
Merged, thanks very much!

**Pull request merged**